numpy/testing/tests/test_utils.py

```diff
@@ -1081,6 +1081,12 @@ def test_simple(self):
 
         assert_raises(AssertionError,
                       lambda: assert_string_equal("foo", "hello"))
+        
+    def test_regex(self):
+        assert_string_equal("a+*b", "a+*b")
+
+        assert_raises(AssertionError,
+                      lambda: assert_string_equal("aaa", "a+b"))
```

Comment:
Might be a bit less awkward to use a context manager (available from assert_raises or the pytest equivalent). The docstring for `assert_string_equal` also claims to return the `diff` between strings--not sure if we need to test that case with special regex chars.

Reply:
That said, the `lambda` form used here is consistent with the rest of the file.

Reply:
The tests function is pretty old (2009), so could use a rewrite sometime in the future. The fixed code dates back to 2007 ...